crates/pop-parachains/src/templates.rs

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: GPL-3.0
+use strum::{
+	EnumMessage as EnumMessageT, EnumProperty as EnumPropertyT, VariantArray as VariantArrayT,
+};
+use strum_macros::{Display, EnumMessage, EnumProperty, EnumString, VariantArray};
+use thiserror::Error;
+
+#[derive(Clone, Default, Debug, Display, EnumMessage, EnumString, Eq, PartialEq, VariantArray)]
+pub enum Provider {
+	#[default]
+	#[strum(
+		ascii_case_insensitive,
+		message = "Pop",
+		detailed_message = "An all-in-one tool for Polkadot development."
+	)]
+	Pop,
+	#[strum(
+		ascii_case_insensitive,
+		message = "Parity",
+		detailed_message = "Solutions for a trust-free world."
+	)]
+	Parity,
+}
+
+impl Provider {
+	pub fn providers() -> &'static [Provider] {
+		Provider::VARIANTS
+	}
+
+	pub fn name(&self) -> &str {
+		self.get_message().unwrap_or_default()
+	}
+
+	pub fn default_template(&self) -> Template {
+		match &self {
+			Provider::Pop => Template::Base,
+			Provider::Parity => Template::ParityContracts,
+		}
+	}
+
+	pub fn description(&self) -> &str {
+		self.get_detailed_message().unwrap_or_default()
+	}
+
+	pub fn templates(&self) -> Vec<&Template> {
+		Template::VARIANTS
+			.iter()
+			.filter(|t| t.get_str("Provider") == Some(self.to_string().as_str()))
+			.collect()
+	}
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Config {
+	pub symbol: String,
+	pub decimals: u8,
+	pub initial_endowment: String,
+}
+
+#[derive(
+	Clone,
+	Debug,
+	Default,
+	Display,
+	EnumMessage,
+	EnumProperty,
+	EnumString,
+	Eq,
+	PartialEq,
+	VariantArray,
+)]
+pub enum Template {
+	// Pop
+	#[default]
+	#[strum(
+		serialize = "base",
+		message = "Standard",
+		detailed_message = "A standard parachain",
+		props(Provider = "Pop", Repository = "r0gue-io/base-parachain")
+	)]
+	Base,
+	// Parity
+	#[strum(
+		serialize = "cpt",
+		message = "Contracts",
+		detailed_message = "Minimal Substrate node configured for smart contracts via pallet-contracts.",
+		props(Provider = "Parity", Repository = "paritytech/substrate-contracts-node")
+	)]
+	ParityContracts,
+	#[strum(
+		serialize = "fpt",
+		message = "EVM",
+		detailed_message = "Template node for a Frontier (EVM) based parachain.",
+		props(Provider = "Parity", Repository = "paritytech/frontier-parachain-template")
+	)]
+	ParityFPT,
```

Comment:
Like discussed before, We included `ParityFPT` here despite it being unmaintained. Does that motivate to also include EPT? 
```suggestion
	ParityFPT,
    ParityEPT,
```

Reply:
Good point. But people is using this templates, I'd suggest to keep them at least until we have the Pop contracts and evm templates ready.

Reply:
Maybe we can update the description with a [deprecated] suffix.

Reply:
But still in the Parity repo, and they haven't mark it as deprecated

Reply:
Not been updated in ages tho, so providing that as an option could perhaps be seen as an endorsement.

Reply:
We are currently showing a warning message after generating a parachain:
```
NOTE: the resulting parachain is not guaranteed to be audited or reviewed for security vulnerabilities. ──────────────────────────────────────────────────────╮
│                                                                                                                                                                │
│  Please consult the source repository at https://github.com/paritytech/substrate-contracts-node to assess production suitability and licensing restrictions.
```

I think with this is enough, but I can add `deprecated` in the help description of the Parity templates if you both think is needed

Reply:
Only downside is that this is only shown after the parachain is generated rather than before.

I'm happy to leave as is and revisit separately.

Reply:
Added a small label in the --help:
` [PROVIDER]  Template provider. Options are pop or parity (deprecated). [default: pop]`
And in user prompt:
```
 Select a template provider: 
│  ○ Pop 
│  ● Parity (Solutions for a trust-free world. 2 available option(s) [deprecated])
```